measure-element/js/scripts.js

```diff
@@ -0,0 +1,80 @@
+/* Copied from our projects */
+const debounce = (callback, delay) => {
+    let timeout;
+
+    return () => {
+        if (timeout) {
+            clearTimeout(timeout);
+        }
+        timeout = setTimeout(callback, delay);
+    };
+};
+
+
+
+
+
+/**
+* Measures an elements dimensions on init and window resize (after debounce), calls the provided
+* callback function on change.
+* @param {HTMLElement} element              Element to measure
+* @param {function} callback                Function to call with the element's dimensions
+*                                           whenever they change. Argument is an instance of
+*                                           DOMRect (see https://developer.mozilla.org/en-US/docs/Web/API/DOMRect)
+* @param {number|false} resizeDebounceInMs  Debounce after a window resize in ms; use 0 for no
+*                                           debounce at all (callback is directly called). Use
+*                                           false if callback should not be called on window resize.
+* @returns {function}                       Function that tears down all listeners (currently
+                                            window resize if resizeDebounceInMs is not false)
+*/
+const measureElement = ({ element, callback, resizeDebounceInMs = 200 } = {}) => {
+    if (!element instanceof HTMLElement) {
+        throw new Error(`Argument \'element\' is required and must be an instance of an HTMLElement; you passed ${JSON.stringify(element)} instead.`);
```

Comment:
Warum die `\`?

Und warum `JSON.stringify(element)`? Das Element wird in der Konsole besser ausgegeben, wenn man das Objekt übergibt.

Reply:
👍  `\` sind in den Template Literals natürlich unnötig – weg sind sie. 

`JSON.stringify`, weil's ein `Error` ist. Der gibt in der Konsole nur Strings aus und formatiert weder Zahlen noch Objekte. Ohne das gäb's z.B. `[object Object]` aus statt `{ a: 5 }`.